src/entities/nestedJoin/index.ts

```diff
@@ -0,0 +1,270 @@
+import {
+    createPublicClient,
+    decodeAbiParameters,
+    decodeFunctionResult,
+    encodeFunctionData,
+    http,
+} from 'viem';
+import { Address, Hex } from '../../types';
+import { Slippage } from '../slippage';
+import { Token } from '../token';
+import { BALANCER_RELAYER, CHAINS, getPoolAddress } from '../../utils';
+import { Relayer } from '../relayer';
+import { parseNestedJoinArgs } from '../utils/parseNestedJoinArgs';
+import { TokenAmount } from '../tokenAmount';
+import { balancerRelayerAbi, bathcRelayerLibraryAbi } from '../../abi';
+
+export type NestedJoinInput = {
+    amountsIn: {
```

Comment:
This format of user input is based off the discussion we had previously with Daniel?

Reply:
Yes, we agreed on not using `TokenAmount` externally, right? I'm just not sure if we're going to need `decimals` or any other token info here. I think more data requirements will pop up once I start adding complexity, such as duplicate tokens that will require amount manipulation before building the nested calls.